onnxruntime/test/testdata/custom_op_library/custom_op_library.cc

```diff
@@ -120,6 +182,23 @@ struct CustomOpOne : Ort::CustomOpBase<CustomOpOne, KernelOne> {
 
 } c_CustomOpOne;
 
+#ifdef USE_CUDA
+struct CustomOpOneCuda : Ort::CustomOpBase<CustomOpOneCuda, KernelOneCuda> {
+  void* CreateKernel(OrtApi api, const OrtKernelInfo* /* info */) const {
+    return new KernelOneCuda(api);
+  };
+
```

Comment:
I think you can implement a method to return the EP - which should be CUDA. By not doing this, this op is actually running on the CPU EP which is why you are having to manually copy data from the host to the device in your kernel's Compute() method